task/import.go

```diff
@@ -47,18 +49,39 @@ func TaskImport(tctx *TaskContext) (*data.TaskOutput, error) {
 		metadataFilePath, err = saveMetadataFile(egCtx, osSess, playbackID, metadata)
 		return err
 	})
+	// Save source file to our storage
 	eg.Go(func() (err error) {
-		fullPath := videoFileName(playbackID)
+		fullPath = videoFileName(playbackID)
 		videoFilePath, err = osSess.SaveData(egCtx, fullPath, secondaryReader, nil, fileUploadTimeout)
 		if err != nil {
 			return fmt.Errorf("error uploading file=%q to object store: %w", fullPath, err)
+		} else {
+			glog.Infof("Saved file=%s to url=%s", fullPath, videoFilePath)
 		}
 		return nil
 	})
+	// Wait for async goroutines finish to run prepare
 	if err := eg.Wait(); err != nil {
-		// TODO: Delete the uploaded file
+		// TODO: Delete the source file
 		return nil, err
 	}
+	// Download our imported output file
+	fileInfoReader, err := osSess.ReadData(ctx, fullPath)
+	if err != nil {
+		return nil, fmt.Errorf("error reading imported file from output OS path=%s err=%w", fullPath, err)
+	}
+	importedFile, err := readFile(fileInfoReader)
+	if err != nil {
+		return nil, err
+	}
+	fileInfoReader.Body.Close()
```

Comment:
If the function above errs, this will leak a file descriptor/connection to the object storage. Instead of handling all of those corner cases manually when there are branches in the code with an expensive resource, I prefer to just `defer Close()` right after getting the thing and not worry about it. 

There's no problem with keeping the file open a little longer until this function exits and the damage is lower than if we start having accidental leaks, so I just go with the `defer` pattern most of the time.
```suggestion
	}
	defer fileInfoReader.Body.Close()
	importedFile, err := readFile(fileInfoReader)
	if err != nil {
		return nil, err
	}
```

Bonus: If you create a helper as I suggested above to avoid the need for a code comment, this `defer Close()` would already close the file ASAP, and it would only return the `importedFile` arg which the caller themselves would `defer Close()` instead. It's the most optimal "file handle lifecycle" with the safety of `defer` handling the closing.